_studio/mfx_lib/encode_hw/mjpeg/src/mfx_mjpeg_encode_hw_utils.cpp

```diff
@@ -210,14 +210,14 @@ mfxStatus ExecuteBuffers::Init(mfxVideoParam const *par, mfxEncodeCtrl const * c
         if (m_payload_base.length + payloadSize > m_payload_base.maxLength)
         {
             mfxU8* data = new mfxU8[m_payload_base.length + payloadSize];
-            memcpy_s(data, m_payload_base.length, m_payload_base.data, m_payload_base.length);
+            std::copy(m_payload_base.data, m_payload_base.data + m_payload_base.length, data);
             delete[] m_payload_base.data;
             m_payload_base.data = data;
             m_payload_base.maxLength = m_payload_base.length + payloadSize;
         }
         m_payload_list.resize(1);
         m_payload_list.back().data = m_payload_base.data + m_payload_base.length;
-        memcpy_s(m_payload_list.back().data, payloadSize, &m_app14_data, payloadSize);
+        m_payload_list.back().data = (mfxU8 *)&m_app14_data;
```

Comment:
no, here you did pointer assignment, but not copy, you need to fairly copy the array:
```
std::copy(m_app14_data, m_app14_data + payloadSize, m_payload_list.back().data);
```